### DISCUSSION TITLE
incremental pca

### DISCUSSION THREAD

**Discussion by san-vak** [vaexio/vaex#2271] [General]
Hi!

does incremental pca apply standardscaler before applying the pca?

**Comment by JovanVeljanoski** [comment]
It is the same as in `scikit-learn`

**Reply by san-vak** [reply]
thanks for your prompt response

original scikit-learn take a "whiten" parameter which is False by default. so can i define whiten parameter in "vaex.ml.PCAIncremental" like this:
```
pca = vaex.ml.PCAIncremental(n_components=200, whiten=True, features=features, batch_size=30_000)
 
pca.fit(df, progress=True).jit_numba()
```

**Reply by JovanVeljanoski** [reply]
The `whiten` parameter is supported. The `PCAIncremental` is a wrapper around `scikit-learn`s incremental pca.

**Reply by JovanVeljanoski** [reply]
It generally performs better if you have LOTS of columns (wide dataframe). For a taller dataframe , the `PCA` by Vaex is faster.

**Reply by san-vak** [reply]
thank you so mcuh
may i ask one more question
is there anyway to get the attributes of the original `scikit-learn` like `explained_variance_` using vaex wrapper?

**Reply by JovanVeljanoski** [reply]
yes, just get it normally like in my example on the other issue you opened. The PCA object (incremental or not) has those attributes.

**Reply by san-vak** [reply]
thank you so much
